src/powerbox/powerbox.py

```diff
@@ -157,52 +162,65 @@ def __init__(
             self.pk = pk
 
         self.ensure_physical = ensure_physical
-        self.Ntot = self.N**self.dim
+        self.Ntot = np.prod(self.N)
 
         self.seed = seed
 
-        if N % 2 == 0:
-            self._even = True
-        else:
-            self._even = False
+        self._even = [N_i % 2 == 0 for N_i in self.N]
 
-        self.n = N + 1 if self._even else N
+        self.n = [N_i + 1 if N_i % 2 == 0 else N_i for N_i in self.N]
 
         # Get the grid-size for the final real-space box.
-        self.dx = float(boxlength) / N
+        self.dx = np.array(self.boxlength) / np.array(self.N)
 
     def k(self):
         """The entire grid of wavenumber magitudes."""
-        return _magnitude_grid(self.kvec, self.dim)
+        kval = np.meshgrid(*self.kvec, indexing="ij")
+        kmode = np.sqrt(np.sum([k_i**2 for k_i in kval], axis=0))
+        return kmode
 
     @property
     def kvec(self):
-        """The vector of wavenumbers along a side."""
-        return self.fftbackend.fftfreq(self.N, d=self.dx, b=self.fourier_b)
+        """The vector of wavenumbers along each side."""
+        kvec_arr = ()
+        for i in range(self.dim):
+            kvec_i = self.fftbackend.fftfreq(self.N[i], d=self.dx[i], b=self.fourier_b)
+            kvec_arr += (kvec_i,)
```

Comment:
this could probably just be a comprehension: 

```python
kvecarr = tuple(self.fftbackend.fftfreq(self.N[i], d=self.dx[i], b=self.fourier_b) for i in range(self.dim))
```